qiita_db/analysis.py

```diff
@@ -165,6 +169,71 @@ def create(cls, owner, name, description, parent=None, from_default=False):
         a_id = conn_handler.execute_queue(queue)[0]
         return cls(a_id)
 
+    @classmethod
+    def delete(cls, _id):
```

Comment:
Decided to leave it as an issue https://github.com/biocore/qiita/issues/1176 so the solution is complete when implemented.